DEVELOPMENT.md

```diff
@@ -0,0 +1,74 @@
+# Development
+
+This doc outlines how to build `common-instancetypes` and which dependencies are required.
+
+## Requirements
+
+A `Makefile` with common operations is supplied with this repository.
+
+The following tools are required by each `Makefile` target:
+
+* `make`
+  * `make lint`
+    * [yamllint](https://github.com/adrienverge/yamllint)
+    * [shellcheck](https://github.com/koalaman/shellcheck)
+  * `make generate`
+    * [kustomize](https://kustomize.io/)
+    * [yq](https://github.com/mikefarah/yq)
+  * `make readme`
+    * [yq](https://github.com/mikefarah/yq)
+  * `make schema`
+    * [openapi2jsonschema](https://github.com/instrumenta/openapi2jsonschema)
+  * `make validate`
+    * [kustomize](https://kustomize.io/)
+    * [kubeconform](https://github.com/yannh/kubeconform)
+
+By default `make` runs the `lint`, `validate`, `readme` and `check` targets and their dependencies.
+
+## Running Makefile targets within a container
+
+A [`Dockerfile`](./Dockerfile) is available that provides the `openapi2jsonschema`, `shellcheck` and `yamllint` dependencies. The golang based depedencies are vendored in the `./tools` directory. To run them golang is required, which is also provided by the [`Dockerfile`](./Dockerfile).
+
+To build and use this simply set the `COMMON_INSTANCETYPES_CRI` env variable to `docker` or `podman` before running `make build_image`:
+
+```
+$ COMMON_INSTANCETYPES_CRI=docker make build_image
+$ COMMON_INSTANCETYPES_CRI=docker make
+```
+
```

Comment:
What do you think about add an example with bot containers engines? Like:

Docker:
```
$ COMMON_INSTANCETYPES_CRI=docker make build_image
$ COMMON_INSTANCETYPES_CRI=docker make
```
Podman:
```
$ COMMON_INSTANCETYPES_CRI=podman make build_image
$ COMMON_INSTANCETYPES_CRI=podman make
```